src/solution_part1.kt

```diff
@@ -1 +1,106 @@
+# Kotlin Essentials II - Assignment
+---
+The Kotlin assignment has three components:
+- Part I: Practice with functions.
+- Part II: Practice with Anonymous Functions (Optional).
+- Part III: More Practicing (Optional).
+
+> Note: You must save your solution as `solution_part1.kt` for Part I, and `solution_part2.kt` for Part II.
+
+## Instructions
+---
+We'll be working through problems designed to get more practice with the concepts covered during the previous lesson (Variables, Conditionals).
+
+## Part I - Practice with functions.
+---
+### Question 1
+Create a function called `add()` that takes two input numbers and returns the sum of these numbers.
+```sh
+fun main() {
+    println(add(10,12))
+}
+Output: 22
+
+### Answer 1
+fun add(num1:Int, num2:Int):Int{
+    val num1: Int =10
+    val num2: Int =12
+    println (add(num1 + num2)
+}
+```
+### Question 2
+Create a function called `fullName()` that takes three inputs strings which are first, middle, and last name, and returns the full name.
+```sh
+fun main() {
+    println(fullName("Ahmed","Abdullah","Al Waleed"))
+    fun
+Output: "Ahmed Abdullah Al Waleed"
+```
+
+### Answer 2
+fun fullName(first: String, middle: String, last:String){
+    val first: String = Ahmed
+    val middle: String= Abdullah
+    val last: String = Al Waleed
+            println("$first $middle $last")
+}
+}
+## Part II - Practice with Anonymous Functions (Optional).
+---
+### Question 3
+Create an anonymous function called `maximum()` that takes three number as input and return the largest one of these numbers.
+```sh
+fun main() {
+    println(maximum(10,14,12))
+}
+
+Output: "The largest number is 14"
+```
+### Answer 3
```

Comment:
Nice try, but the `when` expression﻿ is used for only one condition, and this challenge has more than one condition.

you can find more information about `when` in this link: https://kotlinlang.org/docs/control-flow.html#when-expression